app/models/i14y_search.rb

```diff
@@ -53,6 +51,19 @@ def first_page?
 
   protected
 
+  def date_filter_options(filter_options)
+    filter_options[:sort_by_date] = 1 if @sort_by == 'date'
+    filter_options[:min_timestamp] = @since if @since
+    filter_options[:max_timestamp] = @until if @until
+  end
+
+  def included_tags
+    tags = []
+    tags << @affiliate.tag_filters.required.pluck(:tag).sort.join(',') if @affiliate.tag_filters.required.present?
+    tags << @tags if @tags
+    tags.join(',')
```

Comment:
Does the order of tags matter?

Reply:
The order of tags should _not_ matter, but you raise a good question re: why that sort is in there (and in the related expression on ln 35).  For the purposes of this new funcitonality, the goal is just to add the user/searcher driven tag params (`@tags`) to the list of tag filters already present via functionality already in the main codebase (these filters are set globally for all searchers through the admin interface).  I've merely relocated the existing:

`@affiliate.tag_filters.required.pluck(:tag).sort.join(',')`

down into this new method so I can add more terms to it.  I propose leaving this as is for now at least.

Reply:
I think this method needs to be refactored. The `#join` in L62 can be taken out and kept L64; there's no point in calling `#join` twice. Also, in L63, should there be logic to handle `@tags` being a comma-separated string of tags? Is leading/trailing whitespace an issue here?

Reply:
I've removed the unnecessary `sort` here and its partner on ln 35, and also removed the redundant `join`.

To the question of white space and further massaging the comma-separated string, that's already (mostly) handled once this query is shipped over to i14y.

Tag params are sent to `lib/serde.rb` (https://github.com/GSA/i14y/blob/main/lib/serde.rb#L9-L11) then to `lib/ext/string` (https://github.com/GSA/i14y/blob/main/lib/ext/string.rb#L3) where the array is split, stripped, and downcased.